pkg/workload/schemachange/operation_generator.go

```diff
@@ -3941,13 +3928,47 @@ func (og *operationGenerator) createFunction(ctx context.Context, tx pgx.Tx) (*o
 			}
 			return "TRUE", nil //nolint:returnerrcheck
 		},
-	})
+	}
 
+	// TODO(chrisseto): There's no randomization across STRICT, VOLATILE,
+	// IMMUTABLE, STABLE, STRICT, and [NOT] LEAKPROOF. That's likely not relevant
+	// to the schema workload but may become a nice to have.
+	stmt, expectedCode, err := Generate[*tree.CreateRoutine](og.params.rng, og.produceError(), []GenerationCase{
+		// 1. Nothing special, fully self contained function.
+		{pgcode.SuccessfulCompletion, `CREATE FUNCTION { UniqueName } (i int, j int) RETURNS VOID LANGUAGE SQL AS $$ SELECT NULL $$`},
+		// 2. 1 or more table or type references spread across parameters, return types, or the function body.
+		{pgcode.SuccessfulCompletion, `CREATE FUNCTION { UniqueName } ({ ParamRefs }) RETURNS { ReturnRefs } LANGUAGE SQL AS $$ SELECT NULL WHERE { BodyRefs } $$`},
+		// 3. Reference a table that does not exist.
+		{pgcode.UndefinedTable, `CREATE FUNCTION { UniqueName } () RETURNS VOID LANGUAGE SQL AS $$ SELECT * FROM "ThisTableDoesNotExist" $$`},
+		// 4. Reference a UDT that does not exist.
+		{pgcode.UndefinedObject, `CREATE FUNCTION { UniqueName } (IN p1 "ThisTypeDoesNotExist") RETURNS VOID LANGUAGE SQL AS $$ SELECT NULL $$`},
+		// 5. Reference an Enum that's in the process of being dropped
+		{pgcode.UndefinedTable, `CREATE FUNCTION { UniqueName } (IN p1 { DroppingEnum }) RETURNS VOID LANGUAGE SQL AS $$ SELECT NULL $$`},
+	}, placeholderMap)
 	if err != nil {
 		return nil, err
 	}
+
+	// We don't necessarily generate unique function names all the time.
+	// Upon a successful completion, let's check to make sure that our
+	// function doesn't exist already (overloads are fine).
+	if expectedCode == pgcode.SuccessfulCompletion {
+		params := util.Map(stmt.Params, func(t tree.RoutineParam) string {
```

Comment:
👋 i can also add `fold` (i think also known as `reduce`) to this utils pkg as well instead of mapping, but figured i would suggest it first before doing (i promise it would be quick)

Reply:
My 2c: I think the `map` implementation is fine. With `fold`, the closure would have to manage the formatting inline, and it's nice to rely on `strings.Join` for that.